File: GuiRunner/ValueConverters/FailedTestCountToTextConverter.cs
using System;
using System.Globalization;
using System.Linq;
using System.Windows.Data;
using FeldSpar.ClrInterop;
using FeldSparGuiCSharp.VeiwModels;

namespace FeldSparGuiCSharp.ValueConverters
{
    public class FailedTestCountToTextConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            var assemblyModel = (TestAssemblyModel) value;

            var testDetailModels =
                from test in assemblyModel.Tests
                where test.Status == TestStatus.Failure
                select 1;
            return testDetailModels.Count().ToString();
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
}

File: FSharpWpf/FailedTestCountToTextConverter.fs
namespace FeldSparGuiFSharp.ValueConverters
open System
open System.Globalization
open System.Windows
open System.Windows.Data
open FeldSpar.Api.Engine.ClrInterop
open FeldSpar.Api.Engine.ClrInterop.ViewModels

type FailedTestCountToTextConverter () = 
    interface IValueConverter with
        member this.Convert ((value:obj), (targetType:Type), (parameter:obj), (culture:CultureInfo)) =
            let assemblyModel = value :?> ITestAssemblyModel

            let failures = 
                query
                 {
                    for test in assemblyModel.Tests do
                    where (test.Status = TestStatus.Failure)
                    select test
                    count
                 }

            failures.ToString() :> obj

        member this.ConvertBack ((value:obj), (targetType:Type), (parameter:obj), (culture:CultureInfo)) =
            failwith "Not Implemented"

    member this.IValueConverter = this :> IValueConverter

    member this.Convert ((value:obj), (targetType:Type), (parameter:obj), (culture:CultureInfo)) = this.IValueConverter.Convert (value, targetType, parameter, culture)
    member this.ConvertBack ((value:obj), (targetType:Type), (parameter:obj), (culture:CultureInfo)) = this.IValueConverter.ConvertBack (value, targetType, parameter, culture)